hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -193,6 +194,10 @@ public HazelcastClientInstanceImpl(ClientConfig config,
             addressProviders.add(new DiscoveryAddressProvider(discoveryService));
         }
 
+        if (Boolean.valueOf(getClientConfig().getProperty(GroupProperty.DISCOVERY_SPI_ENABLED))) {
```

Comment:
Yes, this should work in this context:
`if (clientProperties.getBoolean(GroupProperty.DISCOVERY_SPI_ENABLED)) {`